packages/casting/src/follower-cosmjs.js

```diff
@@ -112,290 +145,442 @@ export const makeCosmjsFollower = (
   const leader =
     typeof leaderOrMaker === 'function' ? leaderOrMaker() : leaderOrMaker;
 
+  const tendermintClientPs = new Map();
+  /**
+   * @param {string} endpoint
```

Comment:
Can the return be typed? My TS-foo isn't great, but I think it would be something like this (modulo typedefs):
```suggestion
   * @param {string} endpoint
   * @returns {Promise<{abciInfo: () => {lastBlockHeight: number}, abciQuery: ({path?: string, data?: string, height?: number, prove?: boolean}) => {code: any, value: any, log: any}}>}
```

Reply:
It can be, but in this case, the return type is safely inferred. I’m inclined to keep internal function return types implicit. They do get checked on the caller side.